src/timearray.jl

```diff
@@ -199,16 +199,20 @@ function getindex{T,N,D}(ta::TimeArray{T,N,D}, d::D)
     idxs = searchsorted(ta.timestamp, d)
     length(idxs) == 1 ? ta[idxs[1]] : nothing
 end
- 
+
 # multiple dates
 function getindex{T,N,D}(ta::TimeArray{T,N,D}, dates::Vector{D})
     dates = sort(dates)
     idxs, _ = overlaps(ta.timestamp, dates)
     ta[idxs]
 end #getindex
 
+# StepRange{Date,...}
 getindex{T,N,D}(ta::TimeArray{T,N,D}, r::StepRange{D}) = ta[collect(r)]
 
+# StepRange{Int,Int}
+getindex{T,N}(ta::TimeArray{T,N}, r::StepRange{Int,Int}) = ta[collect(r)]
```

Comment:
Any `Integer` subtype and any integer-valued array can be used for indexing; it needn't be limited to `StepRange`s. In this case we probably want to limit indexing to one-dimensional collections. So this could be written more generally as
```julia
getindex{T,N,R<:Integer}(ta::TimeArray{T,N}, r::AbstractVector{R}) = ta[r]
```
Also note that it doesn't need to be `collect`ed for indexing.